shub/utils.py

```diff
@@ -200,15 +237,12 @@ def _get_dependency_name():
 
 
 def _get_dependency_version(name):
-    if isdir('.git'):
-        return pwd_git_version()
-    elif isdir('.hg'):
-        return pwd_hg_version()
-    elif isdir('.bzr'):
-        return pwd_bzr_version()
-
-    version = _last_line_of(run('python setup.py --version'))
-    return "%s-%s" % (name, version)
+    # XXX: Why is the name included in the version? Keeping it that way in case
+    #      there was a good reason behind it
+    if any(os.path.isdir(dirname) for dirname in ('.git', '.hg', '.bzr')):
+        return pwd_version()
+    else:
+        return "%s-%s" % (name, pwd_version())
```

Comment:
This returns strings like "lxml-3.4.4" as version, despite "lxml" already being save in the egg's 'name' attribute on Dash. [There's even a test that ensures this.](https://github.com/scrapinghub/shub/blob/master/tests/test_utils.py#L24) I've talked to @josericardo and he says this was copied from some old legacy deploy scripts. Any objections to dropping it? I.e. make it return "3.4.4" as version (with the name still saved as 'name' attribute)? (/cc @dangra @pablohoffman)